### PR TITLE
ec2_metadata_facts: fixing generated doc render

### LINKED ISSUE #249: <RoleName> doesn't render in the generated html docs
##### SUMMARY
https://docs.ansible.com/ansible/latest/collections/amazon/aws/ec2_metadata_facts_module.html

See everything in the `ansible_ec2_iam_security_credentials_ ` section. `<role-name>` must be getting parsed out at some point. Very confusing.

##### ISSUE TYPE
- Documentation Report

##### ANSIBLE VERSION

```paste below
ansible 2.10.4
```

### PR DESCRIPTION
**Pull request by mandar242**
##### SUMMARY

Fixing the rendering of generated docs for ec2_metadata_facts as items within < > tags are being parsed out.
Fixes #249 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ec2_metadata_facts

### CODE REVIEW

File: plugins/modules/ec2_metadata_facts.py [hunk 1]
```diff
@@ -278,87 +278,87 @@
             description: Metrics; no longer available.
             type: str
             sample: ""
-        ansible_ec2_network_interfaces_macs_<mac address>_device_number:
+        ansible_ec2_network_interfaces_macs__mac_address_device_number:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:281]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_device_number:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 2]
```diff
@@ -278,87 +278,87 @@
...
             description:
                 - The unique device number associated with that interface. The device number corresponds to the device name;
                   for example, a device-number of 2 is for the eth2 device.
                 - This category corresponds to the DeviceIndex and device-index fields that are used by the Amazon EC2 API and the EC2 commands for the AWS CLI.
             type: str
             sample: "0"
-        ansible_ec2_network_interfaces_macs_<mac address>_interface_id:
+        ansible_ec2_network_interfaces_macs__mac_address_interface_id:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:288]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_interface_id:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 3]
```diff
@@ -278,87 +278,87 @@
...
             description: The elastic network interface ID.
             type: str
             sample: "eni-12345678"
-        ansible_ec2_network_interfaces_macs_<mac address>_ipv4_associations_<ip address>:
+        ansible_ec2_network_interfaces_macs__mac_address_ipv4_associations_ip_address:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:292]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_ipv4_associations_ip_address:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 4]
```diff
@@ -278,87 +278,87 @@
...
             description: The private IPv4 addresses that are associated with each public-ip address and assigned to that interface.
             type: str
             sample: ""
-        ansible_ec2_network_interfaces_macs_<mac address>_ipv6s:
+        ansible_ec2_network_interfaces_macs__mac_address_ipv6s:
             description: The IPv6 addresses associated with the interface. Returned only for instances launched into a VPC.
             type: str
             sample: ""
-        ansible_ec2_network_interfaces_macs_<mac address>_local_hostname:
+        ansible_ec2_network_interfaces_macs__mac_address_local_hostname:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:300]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_local_hostname:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 5]
```diff
@@ -278,87 +278,87 @@
...
             description: The interface's local hostname.
             type: str
             sample: ""
-        ansible_ec2_network_interfaces_macs_<mac address>_local_ipv4s:
+        ansible_ec2_network_interfaces_macs__mac_address_local_ipv4s:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:304]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_local_ipv4s:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 6]
```diff
@@ -278,87 +278,87 @@
...
             description: The private IPv4 addresses associated with the interface.
             type: str
             sample: ""
-        ansible_ec2_network_interfaces_macs_<mac address>_mac:
+        ansible_ec2_network_interfaces_macs__mac_address_mac:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:308]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_mac:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 7]
```diff
@@ -278,87 +278,87 @@
...
             description: The instance's MAC address.
             type: str
             sample: "00:11:22:33:44:55"
-        ansible_ec2_network_interfaces_macs_<mac address>_owner_id:
+        ansible_ec2_network_interfaces_macs__mac_address_owner_id:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:312]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_owner_id:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 8]
```diff
@@ -278,87 +278,87 @@
...
             description:
                 - The ID of the owner of the network interface.
                 - In multiple-interface environments, an interface can be attached by a third party, such as Elastic Load Balancing.
                 - Traffic on an interface is always billed to the interface owner.
             type: str
             sample: "01234567890"
-        ansible_ec2_network_interfaces_macs_<mac address>_public_hostname:
+        ansible_ec2_network_interfaces_macs__mac_address_public_hostname:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:319]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_public_hostname:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 9]
```diff
@@ -278,87 +278,87 @@
...
             description:
                 - The interface's public DNS (IPv4). If the instance is in a VPC,
                   this category is only returned if the enableDnsHostnames attribute is set to true.
             type: str
             sample: "ec2-1-2-3-4.compute-1.amazonaws.com"
-        ansible_ec2_network_interfaces_macs_<mac address>_public_ipv4s:
+        ansible_ec2_network_interfaces_macs__mac_address_public_ipv4s:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:325]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_public_ipv4s:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 10]
```diff
@@ -278,87 +278,87 @@
...
             description: The Elastic IP addresses associated with the interface. There may be multiple IPv4 addresses on an instance.
             type: str
             sample: "1.2.3.4"
-        ansible_ec2_network_interfaces_macs_<mac address>_security_group_ids:
+        ansible_ec2_network_interfaces_macs__mac_address_security_group_ids:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:329]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_security_group_ids:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 11]
```diff
@@ -278,87 +278,87 @@
...
             description: The IDs of the security groups to which the network interface belongs. Returned only for instances launched into a VPC.
             type: str
             sample: "sg-01234567,sg-01234568"
-        ansible_ec2_network_interfaces_macs_<mac address>_security_groups:
+        ansible_ec2_network_interfaces_macs__mac_address_security_groups:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:333]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_security_groups:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 12]
```diff
@@ -278,87 +278,87 @@
...
             description: Security groups to which the network interface belongs. Returned only for instances launched into a VPC.
             type: str
             sample: "secgroup1,secgroup2"
-        ansible_ec2_network_interfaces_macs_<mac address>_subnet_id:
+        ansible_ec2_network_interfaces_macs__mac_address_subnet_id:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:337]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_subnet_id:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 13]
```diff
@@ -278,87 +278,87 @@
...
             description: The ID of the subnet in which the interface resides. Returned only for instances launched into a VPC.
             type: str
             sample: "subnet-01234567"
-        ansible_ec2_network_interfaces_macs_<mac address>_subnet_ipv4_cidr_block:
+        ansible_ec2_network_interfaces_macs__mac_address_subnet_ipv4_cidr_block:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:341]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_subnet_ipv4_cidr_block:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 14]
```diff
@@ -278,87 +278,87 @@
...
             description: The IPv4 CIDR block of the subnet in which the interface resides. Returned only for instances launched into a VPC.
             type: str
             sample: "10.0.1.0/24"
-        ansible_ec2_network_interfaces_macs_<mac address>_subnet_ipv6_cidr_blocks:
+        ansible_ec2_network_interfaces_macs__mac_address_subnet_ipv6_cidr_blocks:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:345]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_subnet_ipv6_cidr_blocks:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 15]
```diff
@@ -278,87 +278,87 @@
...
             description: The IPv6 CIDR block of the subnet in which the interface resides. Returned only for instances launched into a VPC.
             type: str
             sample: ""
-        ansible_ec2_network_interfaces_macs_<mac address>_vpc_id:
+        ansible_ec2_network_interfaces_macs__mac_address_vpc_id:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:349]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_vpc_id:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 16]
```diff
@@ -278,87 +278,87 @@
...
             description: The ID of the VPC in which the interface resides. Returned only for instances launched into a VPC.
             type: str
             sample: "vpc-0123456"
-        ansible_ec2_network_interfaces_macs_<mac address>_vpc_ipv4_cidr_block:
+        ansible_ec2_network_interfaces_macs__mac_address_vpc_ipv4_cidr_block:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:353]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_vpc_ipv4_cidr_block:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 17]
```diff
@@ -278,87 +278,87 @@
...
             description: The IPv4 CIDR block of the VPC in which the interface resides. Returned only for instances launched into a VPC.
             type: str
             sample: "10.0.0.0/16"
-        ansible_ec2_network_interfaces_macs_<mac address>_vpc_ipv4_cidr_blocks:
+        ansible_ec2_network_interfaces_macs__mac_address_vpc_ipv4_cidr_blocks:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:357]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_vpc_ipv4_cidr_blocks:
```

File: plugins/modules/ec2_metadata_facts.py [hunk 18]
```diff
@@ -278,87 +278,87 @@
...
             description: The IPv4 CIDR block of the VPC in which the interface resides. Returned only for instances launched into a VPC.
             type: str
             sample: "10.0.0.0/16"
-        ansible_ec2_network_interfaces_macs_<mac address>_vpc_ipv6_cidr_blocks:
+        ansible_ec2_network_interfaces_macs__mac_address_vpc_ipv6_cidr_blocks:
```
Comment (alinabuzachis) [inline plugins/modules/ec2_metadata_facts.py:361]:
```suggestion
        ansible_ec2_network_interfaces_macs_mac_address_vpc_ipv6_cidr_blocks:
```

**Review by alinabuzachis — COMMENTED** [review]
Thank you @mandar242. I left some comments.

File: plugins/modules/ec2_metadata_facts.py (hunk 2 above)
Comment (mandar242) [reply plugins/modules/ec2_metadata_facts.py]: Thank you for review @alinabuzachis.

**Comment by mandar242** [comment]
> Thank you @mandar242. I left some comments.

Thank you for the review @alinabuzachis.

**Comment by tremble** [comment]
recheck

**Review by tremble — APPROVED** [review]

**Review by alinabuzachis — APPROVED** [review]

**Comment by tremble** [comment]
recheck

**Pull request merged**